### ISSUE TITLE
TRPEndpoint not shown in the interface, even though it is set.

### ISSUE THREAD

**Issue by ping-localhost** [trisacrypto/envoy#246]
At `/about` the `TRP Endpoint` is shown as `N/A`, even though it is set in the env. In fact, `TRPEndpoint` seems to not be used anywhere in the code. 

![image](https://github.com/user-attachments/assets/e17c6335-9ea2-47cf-8ede-e953b7b8f76f)

**Comment by daniellemaxwell** [comment]
@ping-localhost The `N/A` value for `TRP Endpoint` on the `About` page is currently a hardcoded value. We will remove the need to have separate `TRISA` AND `TRP` endpoints soon. Once the changes are merged, we'll follow up and let you know.

**Comment by ping-localhost** [comment]
@daniellemaxwell Has this change been merged already and released?

**Comment by bbengfort** [comment]
> @daniellemaxwell Has this change been merged already and released? We are seeing the following:
> 
> ```json
> {"level":"warn","path":"/","ser_name":"trp","version":"0.27.1-beta.33 (a8a5eb7)","method":"GET","resp_time":0.161824,"resp_bytes":83,"status":404,"client_ip":"87.236.176.241","request_id":"01JH50VGHEM30E9VQDAAWZ67ND","severity":"WARNING","time":"2025-01-09T07:53:23Z","message":"trp GET / 404"}
> ```
> 
> A `404` does not seem like something we want. The node switched to TRISA, but we do need to be able to get TRP messages.

@ping-localhost unfortunately the fix is not merged in quite yet; we need to move to a system that has only one endpoint for everything rather than 3 in order to make the travel addresses work; but that came with a few unseen complications. Are you using the webhook? I can wire something up that would have incoming messages get passed to the webhook; but they might not appear in the UI or in the API; would that work as a temporary solution for you to start receiving messages right away?

**Comment by ping-localhost** [comment]
We are currently using the API to fetch all the data we need (so it missing from the API would be a deal breaker right now). The webhook is not being used.

**Comment by bbengfort** [comment]
Alright - let me see if I can get something together for you next week - I don't want you to be blocked by the one port thing!

**Comment by ping-localhost** [comment]
We still cannot receive TRP messages. Any ETA, since we are slowly being forced to look at other solutions.

**Comment by bbengfort** [comment]
@ping-localhost quick update - the receive TRP messages functionality is in this PR: https://github.com/trisacrypto/envoy/pull/270. It should be merged in the next day or so with a release to follow so that you can start testing the functionality.

**Comment by bbengfort** [comment]
Note that this PR is dependent on this TRISA PR: https://github.com/trisacrypto/trisa/pull/183 (as well as some other changes). If you're looking for a TRP client implementation in Go; the TRISA v1.6.0 release contains that package.

**Comment by bbengfort** [comment]
@ping-localhost the inquiry endpoint is now implemented so that you can receive TRP inquiries; the resolution endpoint (for callbacks to the messages you send) and confirmation endpoint (for concluding a transaction) are still TBD - but I'd like to start testing the TRP inquiry endpoint first since I can't test it with 21 analytics. Would you be able to assist with this?

**Comment by ping-localhost** [comment]
> [@ping-localhost](https://github.com/ping-localhost) the inquiry endpoint is now implemented so that you can receive TRP inquiries; the resolution endpoint (for callbacks to the messages you send) and confirmation endpoint (for concluding a transaction) are still TBD - but I'd like to start testing the TRP inquiry endpoint first since I can't test it with 21 analytics. Would you be able to assist with this?

Is this available in v0.30.0?

**Comment by bbengfort** [comment]
> We can definitely try! Is this available in v0.30.0?

Yes! You should be able to Enable TRP and specify the TRP bind address and make discoverability requests to `/identity`, `/extensions`, `/version`, and `/uptime`. And you can initiate transfers at `/transfers`

**Comment by chris-okuda** [comment]
Closing this; very old and it appears to be completed.
